src/org/psnbtech/TetrisKeyAdapter.java

```diff
@@ -0,0 +1,163 @@
+package org.psnbtech;
+
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+public class TetrisKeyAdapter implements KeyListener {
+
+	private Tetris tetris;
+
+	public TetrisKeyAdapter(Tetris tetris) {
+		tetris = this.tetris;
```

Comment:
This is an amazing bug. 
I can't figure out how the app works with this line of code here.

Also any unit test of this class would've uncovered this bug.

Reply:
It doesn't work. It just opens and then throws a null pointer if I try to do anything.
I figured that the bug was in how I set up the class, I'm just not sure what I did wrong.

Reply:
The bug is literally on this line.

I won't elaborate further.